pkg/kubecost/allocation.go

```diff
@@ -1347,13 +1336,13 @@ func (a *Allocation) getIdleKey(options *AllocationAggregationOptions) (string,
 		// Key allocations to ProviderId to match against node
 		idleKey = a.Properties.ProviderID
 		if idleKey == "" {
-			return idleKey, fmt.Errorf("ProviderId is not set")
+			return UnmountedSuffix, fmt.Errorf("ProviderId is not set")
```

Comment:
Do we know that this will always be the case? Isn't it possible (e.g. in an error state) to be missing a provider ID and not because the allocation is unmounted?

Reply:
Another way of phrasing this is: how can we be certain that everything missing a ProviderID (or a Cluster in the other case) is, in fact, an Unmounted PV?

Reply:
I am not married to using the unmountedSuffix here, I could just as well switch it back to returning "", it just seemed more accurate given the the only instance of missing idleKey I have seen

Reply:
If you don't mind, then let's switch back to `""`. Just because at this moment that is the case, doesn't mean it will always be the case, and I don't see another reason to believe that "no provider ID" should always imply "unmounted PV".

Reply:
Switched back now